src/Zobrist.cpp

```diff
@@ -25,28 +25,30 @@ std::array<std::uint64_t, FastBoard::MAXSQ>                    Zobrist::zobrist_
 std::array<std::array<std::uint64_t, FastBoard::MAXSQ * 2>, 2> Zobrist::zobrist_pris;
 std::array<std::uint64_t, 5>                                   Zobrist::zobrist_pass;
 
+std::uint64_t getRandHash(Random& rng) {
```

Comment:
Let's add this to Random() instead. Additionally adding a randuint64 that takes an argument (for consistency with the others) is trickier to get right but we can just rely on the C++11 random lib and pass our own gen as UniformRandomBitGenerator.

Reply:
Moved the current zero arg implementation to Random, this passes the current tests.

I'd prefer to do a 2nd check in that changes the random algorithm and tests (so that it's clean what changed the test and doesn't cast doubt on bad casts in this commit)